packages/core/tests/repository/cache.repository.test.ts

```diff
@@ -0,0 +1,212 @@
+        const cacheDataSource = new InMemoryDataSource<Book>();
+        const validator = new DefaultObjectValidator();
+
+        repository = new CacheRepository<Book>(
+            mainDataSource,
+            mainDataSource,
+            mainDataSource,
+            cacheDataSource,
+            cacheDataSource,
+            cacheDataSource,
+            validator,
+        );
+    });
+
+    describe('get', () => {
+        it('should throw an error when a non-supported operation is given', () => {
+            const keyQuery = getDefaultKeyQuery();
+            const notSupportedOperation = getNotSupportedOperation();
+
+            const result = repository.get(keyQuery, notSupportedOperation);
+
+            expect(result).rejects.toThrow(OperationNotSupportedError);
+        });
+
+        it('should return undefined when the key is not found', async () => {
+            const nonExistingKeyQuery = getMissingKeyQuery()
+            const operation = getDefaultOperation();
+
+            const result = await repository.get(nonExistingKeyQuery, operation);
+
+            expect(result).toBe(undefined);
+        });
+
+        it('should return the expected value when an existing is given', async () => {
+            const query = getDefaultKeyQuery();
+            const operation = getDefaultOperation();
+            const book = getDefaultBook();
+
+            await repository.put(book, query, operation);
+            const result = await repository.get(query, operation);
+
+            expect(result).toBe(book);
+        });
+
+        it('should return the expected value when the key exists in the main datasource', async () => {
+            const query = getDefaultKeyQuery();
+            const operation = getMainOperation();
+            const book = getDefaultBook();
+
+            await repository.put(book, query, operation);
+            const result = await repository.get(query, operation);
+
+            expect(result).toBe(book);
+        });
+
+        it('should return the expected value when the key is already cached', async () => {
+            const query = getDefaultKeyQuery();
+            const operation = getCacheOperation();
+            const book = getDefaultBook();
+
+            await repository.put(book, query, operation);
+            const result = await repository.get(query, operation);
+
+            expect(result).toBe(book);
+        });
+    });
+
+    describe('put', () => {
+        it('should throw an error when a non-supported operation is given', async () => {
+            const book = getDefaultBook();
+            const keyQuery = getDefaultKeyQuery();
+            const notSupportedOperation = getNotSupportedOperation();
+
+            const result = repository.put(book, keyQuery, notSupportedOperation);
+
+            expect(result).rejects.toThrowError(OperationNotSupportedError);
+        });
+
+        it('should return the given value when this is inserted', async () => {
```

Comment:
```suggestion
        it('should return the given value on insertion', async () => {
```

Reply:
Following what Miguel said:
My personal pick of all those options is this -> "should HAVE_EXPECTED_BEHAVIOUR when RACE_CONDITION"

Reply:
In this case `on` also denotes an event; like when you say `onClick` you could call it `whenTheUserClicks`; I think is synonym.